### ISSUE TITLE
TCPObject is broken

### ISSUE THREAD

**Issue by pacomont** [GarageGames/Torque3D#1951]
Client connection to a server is not performed. It could be tested trying to connect to a TelnetDebugger service ( startDebugger() ) witch uses TCPObjects. 

When trying to connect **Net::openConnectTo()** is called, but It exists with handleFd = NetSocket::INVALID;

/// To use the debugger, use dbgSetParameters(port, password); in the console
/// of the server to enable debugger connections.  Then on some other system,
/// start up the app (you don't have to start a game or connect to the
/// server) and exec("common/debugger/debugger.cs"); in the console.  Then use
/// the debugger GUI to connect to the server with the right port and password.

**Comment by Azaezel** [comment]
Can you elaborate further on your reproduction case? Masterserver connectivity and lan both seem functional at present. (Will have to wait a bit for reproduction testing on external network testing this end, I'm afraid.)

**Comment by pacomont** [comment]
Of course, sorry... 
[1] Launch 2 instances of the application.
[2] In the first instance (server) execute in console: **dbgSetParameters(3333,"aa");**
[3] In the second instance (client) execute in console: **startDebugger();**
[4] In second instance press the connect button and complete with the address **"127.0.0.1", port "3333" and password "aa"**...
You'll see that nothing happens, the client will not connect with the server.   

You could trace the execution in the client until TCPObject::connect(), and Net::openConnectTo(address); will return NetSocket::INVALID.
`void TCPObject::connect(const char *address)
{
   NetSocket newTag = Net::openConnectTo(address);
   addToTable(newTag);
}`

Image of 2 instances:
[https://goo.gl/SQhUsj](url)

**Comment by Azaezel** [comment]
Yeah, ip connectivity did get tweaked a bit on the backend to support IPV6: https://github.com/GarageGames/Torque3D/pull/1817 likely an oversight... Gotta say, first I'd even heard of that subsystem, so it'll take some digging if nothing springs out at you (since you're more familiar with that subsystem)

**Comment by pacomont** [comment]
Oh wait, I found the real problem (but not the solution, yet...). Forget previous messages, here is the new case:

Using HTTPObject with the stock example from source code ([link](https://gist.github.com/pacomont/961a32674d638eb000858c5c3107442f)), 

I found that using:
`%feed.get("www.google.com:80", "/ig/api", "weather=Las-Vegas,US"); `
it **works** 

but expecifing ip address directly, it **doesn't work**: 
`%feed.get("216.58.208.228:80", "/ig/api", "weather=Las-Vegas,US"); //it's google.com ip address from Spain`

To use, execute: **testHTTPObject()**, and then **echo(WeatherFeed.buffer);**

Let me see if I can find the cause...

**Comment by jamesu** [comment]
Looks mainly like a typo in the code. Might want to tweak the error handling a bit, I don't have a solid doc for what non-blocking sockets should be returning besides badly documented sample code. Anyone got any good socket docs? ;)
